controlplane/kubeadm/internal/controllers/controller_test.go

```diff
@@ -147,7 +147,7 @@ func TestReconcileReturnErrorWhenOwnerClusterIsMissing(t *testing.T) {
 	g.Expect(result).To(Equal(ctrl.Result{}))
 
 	// calling reconcile should return error
-	g.Expect(env.Delete(ctx, cluster)).To(Succeed())
+	g.Expect(env.CleanupAndWait(ctx, cluster)).To(Succeed())
```

Comment:
If the cache is not updated to reflect the cluster deletion then this test panic. Adding a `CleanupAndWait` makes sure that the test tests exactly the desired thing.

Reply:
Okay now I understand why the test panic'ed and I also understand why that fixes the panic in the test.

I assume this panic can't happen in reality because the code panic'ed because `r.managementCluster` was nil?

Reply:
For this particular test case it panics at the `List` operation of the `GetCRDWithContract` function because the client used here is the `APIReader` which is not provided in the test case. Even if we add the `APIReader` in the test case the test would still panic because `r.managementCluster` is nil.

Reply:
Okay got it. So overall this panic only happens in tests. Thx